modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/BinarySpatialOpTypeBinding.java

```diff
@@ -55,6 +62,18 @@ public Class getType() {
     
     @Override
     public Object getProperty(Object object, QName name) throws Exception {
-        return FESParseEncodeUtil.getProperty((BinarySpatialOperator) object, name);
+        Expression e = FESParseEncodeUtil.getProperty((BinarySpatialOperator) object, name);
+        if (e instanceof Literal && ((Literal) e).getValue() instanceof Geometry) {
+            return new EncoderDelegate() {
+                @Override
+                public void encode(ContentHandler output) throws Exception {                    
+                    Encoder encoder = new Encoder(new org.geotools.gml3.v3_2.GMLConfiguration());
```

Comment:
I can successfully change this line:
Encoder encoder = new Encoder(new org.geotools.gml3.v3_2.GMLConfiguration());
to this line:
Encoder encoder = new Encoder(new FESConfiguration());

Because,as you said, FESConfiguration depends on gml3.v3_2.GMLConfiguration.

But, that still leaves us with this line:
encoder.encode(((Literal) e).getValue(), org.geotools.gml3.v3_2.GML.AbstractGeometry, output);

And I must specify AbstractGeometry, because that is the whole reason of the EncoderDelegate, to explicitly specify that the object needs to be encoded as a GML geometry.

But considering that, as you pointed out, FESConfiguration is already depending on GML3.2 Configuration in geotools. That means that we don't actually have a choice here, in our geotools implementation FES does depend on GML 3.2